db/db_options_test.cc

```diff
@@ -178,6 +178,22 @@ TEST_F(DBOptionsTest, EnableAutoCompactionAndTriggerStall) {
   }
 }
 
+TEST_F(DBOptionsTest, EnableAutoCompactionShouldTriggerCompaction) {
+  Options options;
+  options.create_if_missing = true;
+  options.level0_file_num_compaction_trigger = 1000;
+  Reopen(options);
+  for (int i = 0; i < 3; i++) {
+    ASSERT_OK(Put(ToString(i), ToString(i)));
+    Flush();
+  }
+  ASSERT_EQ("3", FilesPerLevel());
+  ASSERT_OK(
+      dbfull()->SetOptions({{"level0_file_num_compaction_trigger", "3"}}));
+  dbfull()->TEST_WaitForCompact();
+  ASSERT_LT(0, NumTableFilesAtLevel(1));
```

Comment:
nit: I think it will be clear if we make this check
ASSERT_EQ(0, NumTableFilesAtLevel(0))

We always compact all files in L0 so we should expect no files exist in L0 after the auto compaction